vktrace/vktrace_layer/vktrace_lib_pageguard.cpp

```diff
@@ -388,6 +414,29 @@ LONG WINAPI PageGuardExceptionHandler(PEXCEPTION_POINTERS ExceptionInfo) {
     pageguardExit();
     return resultCode;
 }
+#elif defined(ANDROID)
+void PageGuardExceptionHandler(int sig, siginfo_t *si, void *unused)
+{
+    if (sig == SIGSEGV)
+    {
+        VkDeviceSize OffsetOfAddr;
+        PBYTE pBlock;
+        VkDeviceSize BlockSize;
+        PBYTE addr = (PBYTE)si->si_addr;
```

Comment:
Would it be possible to add process for reading memory and capture (CPU needed) data change coming from GPU side?

Reply:
The  siginfo_t struct seems not provide the access type info for the exception. Now we just add the write protect to the mapped memory address, so we can always assume the fault exception is caused by the write access.

So far, we didn't meet a read case on android. Mapping a memory permanently for writing might be a normal case, but mapping a memory permanently for reading might be very rare case.

Reply:
Is there some context info can be used?  of course, it's ok if no such case on Android. That's quite different with Windows titles.